### PR TITLE
Fix bug in changelog generation

### LINKED ISSUE #148: Fix bug in changelog generation
### Description
Currently `merged-changelog<<EOF` appears inside the generated change log of a GitHub release, which shouldn't be the case. This also seems to cause weird formatting. It is unclear if this also happens when the changelog is being written into a file.

### Acceptance criteria
- [ ] The changelog in a GitHub release is generated without `merged-changelog<<EOF` in its content
- [ ] The changelog in a GitHub release is correctly formatted
- [ ] The changelog in a markdown file is generated without `merged-changelog<<EOF` in its content
- [ ] The changelog in a markdown file is correctly formatted

### PR DESCRIPTION
**Pull request by MichaelKora**
Fixes #148

### CODE REVIEW

File: actions/changelog-generate/action.yaml [hunk 1]
```diff
@@ -74,16 +73,11 @@ runs:
           fi
         fi
         echo "fromtag=$setfrom">> $GITHUB_OUTPUT
-        if [ -n "${{ inputs.config }}" ]; then
+        if [  "${{ inputs.config }}" != '']; then
```
Comment (disrupted) [inline actions/changelog-generate/action.yaml:76]: can't you check for non-empty using `-z`?

**Review by disrupted — CHANGES_REQUESTED** [review]

File: .github/workflows/python-poetry-release.yaml [hunk 2]
```diff
@@ -85,7 +85,7 @@ jobs:
 
       - name: Create changelog
         id: build-changelog
-        uses: bakdata/ci-templates/actions/changelog-generate@1.35.0
+        uses: bakdata/ci-templates/actions/changelog-generate@feat/changelog-eof
```
Comment (yannick-roeder) [inline .github/workflows/python-poetry-release.yaml:88]: Todo

File: actions/changelog-generate/action.yaml [hunk 3]
```diff
@@ -74,16 +73,14 @@ runs:
           fi
         fi
         echo "fromtag=$setfrom">> $GITHUB_OUTPUT
-        if [ -n "${{ inputs.config }}" ]; then
-          path="$GITHUB_ACTION_PATH/changelog-config.json)"
+        # if [  "${{ inputs.config }}" != '']; then
+        if [  -z "${{ inputs.config }}" ]; then
```
Comment (yannick-roeder) [inline actions/changelog-generate/action.yaml:76]:
```suggestion
        if [ -z "${{ inputs.config }}" ]; then
```

**Review by yannick-roeder — CHANGES_REQUESTED** [review]

File: .github/workflows/python-poetry-release.yaml (hunk 2 above)
Comment (MichaelKora) [reply .github/workflows/python-poetry-release.yaml:88]: thanks... 
will do it if you think everything else is ok.

File: actions/changelog-generate/action.yaml [hunk 4]
```diff
@@ -74,16 +73,11 @@ runs:
...
+        if [ -z "${{ inputs.config }}" ]; then
+          path="$GITHUB_ACTION_PATH/changelog-config.json"
         else
           path=${{ inputs.config }}
         fi
-        if [ ! -f "$path" ]; then
-          echo "changelog config file $path doesn't exist"
-          exit 1
-        fi
-        echo "path=$path" >> "$GITHUB_OUTPUT"
```
Comment (disrupted) [inline actions/changelog-generate/action.yaml:86]: why did you remove this check and how does the `path` output variable work now?

Comment (MichaelKora) [reply actions/changelog-generate/action.yaml:86]: the check was removed because it was not working as expected. the path works basically the same.. if a config was provided then it will be used; else the one from the action will be used

Comment (disrupted) [reply actions/changelog-generate/action.yaml:86]: what if a config path was provided but there's a mistake and the file doesn't actually exist?
also I believe you still need this last line, otherwise the output variable is not created

Comment (MichaelKora) [reply actions/changelog-generate/action.yaml]: oh good catch...it should be fine..now..also test in PG

File: actions/changelog-generate/action.yaml [hunk 5]
```diff
@@ -73,12 +72,14 @@ runs:
             setfrom=$(git rev-list --max-parents=0 HEAD)
           fi
         fi
+
         echo "fromtag=$setfrom">> $GITHUB_OUTPUT
-        if [ -n "${{ inputs.config }}" ]; then
-          path="$GITHUB_ACTION_PATH/changelog-config.json)"
+        if [ -Z "${{ inputs.config }}" ]; then
```
Comment (yannick-roeder) [inline actions/changelog-generate/action.yaml:77]: Did you test this? When I test it locally with `-Z` instead of `-z` then I get the following error:

```
bash: conditional binary operator expected
```

```suggestion
        if [[ -z "${{ inputs.config }}" ]]; then
```

**Review by yannick-roeder — CHANGES_REQUESTED** [review]

**Review by yannick-roeder — DISMISSED** [review]

File: .github/workflows/python-poetry-release.yaml (hunk 2 above)
Comment (MichaelKora) [reply .github/workflows/python-poetry-release.yaml]: done

**Review by yannick-roeder — APPROVED** [review]

**Review by disrupted — APPROVED** [review]
